server/src/test/java/org/opensearch/common/cache/store/OpenSearchOnHeapCacheTests.java

```diff
@@ -107,7 +112,8 @@ private void assertZeroStats(ImmutableCacheStatsHolder stats) {
     private OpenSearchOnHeapCache<String, String> getCache(
         int maxSizeKeys,
         MockRemovalListener<String, String> listener,
-        boolean pluggableCachesSetting
+        boolean pluggableCachesSetting,
+        boolean useNoopStatsInConfig
```

Comment:
This is here because if we don't draw stats from the tiers themselves, which I don't think we can feasibly do for the reasons above, it doesn't make sense for those tiers to track their stats and use up memory.

Reply:
That's a good point. It should be seamless and users need not know about it.

Reply:
End users of the cache shouldn't have to worry about this parameter in the CacheConfig, as it defaults to false and is set to true automatically by the TSC before it constructs its heap and disk tiers. 

If we removed the parameter, is there another way the tiers could know they were part of the tiered cache, and use NoopStats instead? Checking the feature flag wouldn't be able to tell them that.

Reply:
Maybe the tiers could check`CacheSettings.getConcreteStoreNameSettingForCacheType(cacheType.getSettingPrefix())` to see whether the tiered cache is being used? But this would require them to know about `TieredSpilloverCacheFactory.TIERED_SPILLOVER_CACHE_NAME`, which lives in a module.

Reply:
@peteralfonsi - I think it should get plugged in implicitly, instead of having to explicitly say `useNoopStats`.

Reply:
@msfroh @jainankitk Would it be clearer what's going on if instead we passed an `statsTrackingEnabled` parameter to CacheConfig?